### PR TITLE
Removes RnG from the Ablative Trenchcoat and Armor

### PR DESCRIPTION
**Pull request by Coolrune206**
## What Does This PR Do

Removes the reflect chance from the Ablative Armor and Trenchcoat in favour of a cooldown system. Every four seconds, the Ablative items can fully deflect one laser/energy projectile. This deflection period lasts a decisecond, so that if someone decides to use the really cool lasershot on you or some other type of scatter laser that's all deflected in the first hit.
## Why It's Good For The Game

The Ablative armorpieces previously had an RnG chance to reflect attacks. Combat RnG is generally bad, especially when it's a passive bonus. This keeps the Ablative Armor strong at fighting people with energy weapons while removing the random chance to reflect projectiles. Also, previously the two ablative armors had different reflect chances - I've just flattened them to the same cooldown for now, but am willing to change this if necessary.
## Testing

I shot the shit out of a skrell wearing the Ablative. The first shot was blocked, the rest went through until four seconds had passed. Ditto with the scatter-laser, except all the projectiles in the first shot were reflected.
## Changelog
:cl:
tweak: Changes the Ablative Armor from an RnG reflect chance to a cooldown-based guaranteed reflect
/:cl:

### CODE REVIEW

File: code/modules/clothing/suits/miscellaneous.dm [hunk 1]
```diff
@@ -535,12 +535,18 @@
...
 /obj/item/clothing/suit/hooded/ablative/IsReflect()
 	var/mob/living/carbon/human/user = loc
-	if(prob(hit_reflect_chance) && (user.wear_suit == src))
+	if(user.wear_suit != src)
+		return
+	if(world.time - lastreflect >= 4 SECONDS)
+		lastreflect = world.time
+		return 1
+	if(world.time - lastreflect <= 1)
 		return 1
```
Comment (hal9000PR) [inline code/modules/clothing/suits/miscellaneous.dm:548]: comment for why this is needed please.

File: code/modules/clothing/suits/miscellaneous.dm [hunk 2]
```diff
@@ -535,12 +535,18 @@
...
+	var/lastreflect
 
 /obj/item/clothing/suit/hooded/ablative/IsReflect()
 	var/mob/living/carbon/human/user = loc
-	if(prob(hit_reflect_chance) && (user.wear_suit == src))
+	if(user.wear_suit != src)
+		return
+	if(world.time - lastreflect >= 4 SECONDS)
+		lastreflect = world.time
+		return 1
```
Comment (hal9000PR) [inline code/modules/clothing/suits/miscellaneous.dm:546]: make the `4 SECONDS` a variable please

File: code/modules/clothing/suits/miscellaneous.dm [hunk 3]
```diff
@@ -535,12 +535,18 @@
 	put_on_delay = 4 SECONDS
 	sprite_sheets = list("Vox" = 'icons/mob/clothing/species/vox/suit.dmi',
 						 "Grey" = 'icons/mob/clothing/species/grey/suit.dmi')
-	var/hit_reflect_chance = 50
+	var/lastreflect
```
Comment (hal9000PR) [inline code/modules/clothing/suits/miscellaneous.dm:538]: would you rename this to `last_reflect_time`

**Review by hal9000PR — CHANGES_REQUESTED** [review]
all of these reviews for the trenchcoat and the armour please

File: code/modules/clothing/suits/miscellaneous.dm (hunk 1 above)
Comment (Coolrune206) [reply code/modules/clothing/suits/miscellaneous.dm:548]: fuck I forgot the comments

**Comment by Bm0n** [comment]
Make sure you change the ablative vest as well. Probably also make it a little bit weaker

File: code/modules/clothing/suits/armor.dm [hunk 4]
```diff
@@ -315,12 +315,19 @@
 	item_state = "armor_reflec"
 	blood_overlay_type = "armor"
 	armor = list(MELEE = 5, BULLET = 5, LASER = 75, ENERGY = 50, BOMB = 0, BIO = 0, RAD = 0, FIRE = INFINITY, ACID = INFINITY)
-	var/hit_reflect_chance = 40
+	var/last_reflect_time
+	var/reflect_cooldown = 4 SECONDS
```
Comment (hal9000PR) [inline code/modules/clothing/suits/armor.dm:319]: Would you make this one 5 seconds?

File: code/modules/clothing/suits/armor.dm [hunk 5]
```diff
@@ -315,12 +315,19 @@
...
 
 /obj/item/clothing/suit/armor/laserproof/IsReflect()
 	var/mob/living/carbon/human/user = loc
-	if(prob(hit_reflect_chance) && (user.wear_suit == src))
+	if(user.wear_suit != src)
+		return
```
Comment (hal9000PR) [inline code/modules/clothing/suits/armor.dm:324]: This should return false

File: code/modules/clothing/suits/armor.dm [hunk 6]
```diff
@@ -315,12 +315,19 @@
...
 	armor = list(MELEE = 5, BULLET = 5, LASER = 75, ENERGY = 50, BOMB = 0, BIO = 0, RAD = 0, FIRE = INFINITY, ACID = INFINITY)
-	var/hit_reflect_chance = 40
+	var/last_reflect_time
+	var/reflect_cooldown = 5 SECONDS
 
 /obj/item/clothing/suit/armor/laserproof/IsReflect()
 	var/mob/living/carbon/human/user = loc
-	if(prob(hit_reflect_chance) && (user.wear_suit == src))
+	if(user.wear_suit != src)
+		return 0
```
Comment (Sirryan2002) [inline code/modules/clothing/suits/armor.dm:324]: I hate that these don't use defines :/

**Review by Sirryan2002 — APPROVED** [review]
code looks good

**Review by SteelSlayer — APPROVED** [review]

**Comment by SteelSlayer** [comment]
Needs a deconflict, then it can be merged.

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**